### PR TITLE
Ignore support requests for which user data is removed when retrieving unresolved issues

### PR DESCRIPTION
**Pull request by xavierdecoster**

### CODE REVIEW

File: src/NuGet.SupportRequests.Notifications/SqlQuery.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ internal static class SqlQuery
         internal static string GetUnresolvedIssues(string onCallPagerDutyUserName = null)
         {
             var query =
-                "SELECT I.[CreatedBy], I.[CreatedDate], I.[PackageId], I.[PackageVersion], I.[OwnerEmail], I.[Reason], I.[PackageRegistrationKey], ISNULL(A.[PagerDutyUsername], \'Unassigned\') AS \'AdminPagerDutyUsername\', ISNULL(A.[GalleryUsername], \'Unassigned\') AS \'AdminGalleryUsername\', I.[IssueStatusId] AS \'IssueStatus\' FROM [dbo].[Issues] AS I (NOLOCK) LEFT OUTER JOIN [dbo].[Admins] AS A (NOLOCK) ON I.[AssignedToId] = A.[Key] WHERE I.[IssueStatusId] <> 3 ";
+                "SELECT I.[CreatedBy], I.[CreatedDate], I.[PackageId], I.[PackageVersion], I.[OwnerEmail], I.[Reason], I.[PackageRegistrationKey], ISNULL(A.[PagerDutyUsername], \'Unassigned\') AS \'AdminPagerDutyUsername\', ISNULL(A.[GalleryUsername], \'Unassigned\') AS \'AdminGalleryUsername\', I.[IssueStatusId] AS \'IssueStatus\' FROM [dbo].[Issues] AS I (NOLOCK) LEFT OUTER JOIN [dbo].[Admins] AS A (NOLOCK) ON I.[AssignedToId] = A.[Key] WHERE I.[IssueStatusId] <> 3 AND I.[CreatedBy] IS NOT NULL";
```
Comment (loic-sharma) [inline src/NuGet.SupportRequests.Notifications/SqlQuery.cs]: Please add a comment that if `I.[CreatedBy]` is null then the user data is removed and the support request should be ignored.

File: src/NuGet.SupportRequests.Notifications/Program.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ public class Program
         public static void Main(string[] args)
         {
             var job = new Job();
-            JobRunner.Run(job, args).Wait();
+            JobRunner.Run(job, args).GetAwaiter().GetResult();
```
Comment (cristinamanum) [inline src/NuGet.SupportRequests.Notifications/Program.cs]: Why was this change needed?

Comment (loic-sharma) [reply src/NuGet.SupportRequests.Notifications/Program.cs]: `GetAwaiter().GetResult()` is the proper way for synchronous code to synchronously wait on asynchronous code. `Wait()` will deadlock if used improperly.

Comment (xavierdecoster) [reply src/NuGet.SupportRequests.Notifications/Program.cs]: What @loic-sharma says :) Noticed this particular change in other jobs, so figured it would be good for this one to follow the same approach.

Comment (joelverhagen) [reply src/NuGet.SupportRequests.Notifications/Program.cs]: Both can deadlock in the same contexts. The only difference is how exceptions are wrapped. But both are totally fine to use as a console app entry point. `GetAwaiter().GetResult()` just gives prettier exceptions.
